src/coreclr/nativeaot/Runtime/i386/WriteBarriers.asm

```diff
@@ -238,29 +240,76 @@ DEFINE_CHECKED_WRITE_BARRIER EDX, EBP
 ;; WARNING: Code in EHHelpers.cpp makes assumptions about write barrier code, in particular:
 ;; - Function "InWriteBarrierHelper" assumes an AV due to passed in null pointer will happen at @RhpCheckedLockCmpXchgAVLocation@0
 ;; - Function "UnwindSimpleHelperToCaller" assumes the stack contains just the pushed return address
-;; pass third argument in EAX
-FASTCALL_FUNC RhpCheckedLockCmpXchg
-ALTERNATE_ENTRY RhpCheckedLockCmpXchgAVLocation
+FASTCALL_FUNC RhpCheckedLockCmpXchg, 12
+    mov             eax, [esp+4]
```

Comment:
I wonder if this change and the fact that the caller now has to push the argument to stack have a measurable perf impact.

Reply:
It's not really a change, just a difference between the custom ProjectN ABI and the RyuJIT ABI. This just aligns with CoreCLR and RyuJIT.